crates/core/src/commands/check.rs

```diff
@@ -667,7 +687,18 @@ fn check_pack(
 
     // check header length
     let header_len = PackHeaderRef::from_index_pack(&index_pack).size();
-    let pack_header_len = PackHeaderLength::from_binary(&data.split_off(data.len() - 4))?.to_u32();
+    let pack_header_len = PackHeaderLength::from_binary(&data.split_off(data.len() - 4))
+        .map_err(|err| {
+            RusticError::with_source(
+                ErrorKind::Command,
```

Comment:
Probably `Internal` is better here. Not sure, if it even makes sense to have the `Command` variant. As most of that is also internal.